src/test/java/com/ericsson/ei/controller/TestFreeStyleQueryImpl.java

```diff
@@ -0,0 +1,130 @@
+import org.apache.qpid.util.FileUtils;
+import org.bson.Document;
+import org.jongo.Jongo;
+import org.jongo.MongoCollection;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.io.File;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest
+public class TestFreeStyleQueryImpl {
+
+    private static final Logger LOGGER = (Logger) LoggerFactory.getLogger(TestFreeStyleQueryImpl.class);
+    private static final String inputPath = "src/test/resources/AggregatedObject.json";
+    private static final String REQUEST = "testCaseExecutions.testCase.verdict:PASSED,testCaseExecutions.testCase.id:TC5,id:6acc3c87-75e0-4b6d-88f5-b1a5d4e62b43";
+    private static final String QUERY = "{\"criteria\" :{\"testCaseExecutions.testCase.verdict\":\"PASSED\", \"testCaseExecutions.testCase.id\":\"TC5\" }, \"options\" :{ \"id\": \"6acc3c87-75e0-4b6d-88f5-b1a5d4e62b43\"} }";
+    private static String input;
+    private static final String DB_NAME = "MissedNotification";
+    private static final String DB_COLLECTION = "Notification";
+
+    @Value("${server.port}")
+    private String serverPort;
+
+    @Autowired
+    private ProcessQueryParams unitUnderTest;
+
+    @Before
+    public void setUp() {
+        input = FileUtils.readFileAsString(new File(inputPath));
+        try (MongoClient mongoClient = new MongoClient()) {
+            MongoDatabase db = mongoClient.getDatabase(DB_NAME);
+            com.mongodb.client.MongoCollection<Document> collection = db.getCollection(DB_COLLECTION);
+            final Document dbObjectInput = Document.parse(input);
+            collection.insertOne(dbObjectInput);
+            if (collection.count() != 0) {
+                System.out.println("Data Inserted successfully in both the Collections");
+            }
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage());
+        }
+    }
+
+    @Test
+    public void filterFormParamTest() throws Exception {
+        JSONObject inputJArr = new JSONObject(input);
+        LOGGER.info("The input string is : " + inputJArr.toString());
+        DB db = new MongoClient().getDB(DB_NAME);
+        Jongo jongo = new Jongo(db);
+        MongoCollection aggObjects = jongo.getCollection(DB_COLLECTION);
+        Document one = aggObjects.findOne().as(Document.class);
+        JSONObject json = new JSONObject(one.toJson());
```

Comment:
json is not used here in any assert so it needs to be removed since it seems it is useless.